android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -49,38 +62,64 @@ private void initState() {
     }
 
     private void setSelectedNetworkObserver() {
-        mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork.observe(
-                this, networkInfo -> {
-                    if (networkInfo != null) {
-                        mSelectedNetwork = Utils.getShortNameOfNetwork(networkInfo.chainName);
-                        networkSelectorAdapter.setSelectedNetwork(networkInfo.chainName);
-                    }
-                });
+        if (mMode == NetworkSelectorModel.Mode.DEFAULT_WALLET_NETWORK) {
+            LiveDataUtil.observeOnce(
+                    mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork,
+                    networkInfo -> { updateNetworkSelection(networkInfo); });
+        } else if (mMode == NetworkSelectorModel.Mode.LOCAL_NETWORK_FILTER) {
+            mNetworkSelectorModel.mSelectedNetwork.observe(
+                    this, networkInfo -> { updateNetworkSelection(networkInfo); });
+        }
+    }
+
+    private void updateNetworkSelection(NetworkInfo networkInfo) {
+        if (networkInfo != null) {
+            mSelectedNetwork = Utils.getShortNameOfNetwork(networkInfo.chainName);
+            networkSelectorAdapter.setSelectedNetwork(networkInfo.chainName);
+        }
     }
 
     @Override
     public void onNetworkItemSelected(NetworkInfo networkInfo) {
-        mWalletModel.getCryptoModel().getNetworkModel().setNetworkWithAccountCheck(
-                networkInfo, isSelected -> {
-                    if (!isSelected) {
-                        Toast.makeText(this,
-                                     getString(R.string.brave_wallet_network_selection_error,
-                                             networkInfo.chainName),
-                                     Toast.LENGTH_SHORT)
-                                .show();
-                        networkSelectorAdapter.setSelectedNetwork(mSelectedNetwork);
-                    }
-                    finish();
-                });
+        if (mMode == NetworkSelectorModel.Mode.DEFAULT_WALLET_NETWORK) {
+            mWalletModel.getCryptoModel().getNetworkModel().setNetworkWithAccountCheck(
+                    networkInfo, isSelected -> { updateNetworkUi(networkInfo, isSelected); });
+        } else if (mMode == NetworkSelectorModel.Mode.LOCAL_NETWORK_FILTER) {
+            mNetworkSelectorModel.setNetworkWithAccountCheck(
+                    networkInfo, isSelected -> { updateNetworkUi(networkInfo, isSelected); });
+        }
+    }
+
+    private void updateNetworkUi(NetworkInfo networkInfo, Boolean isSelected) {
+        if (!isSelected) {
+            Toast.makeText(this,
+                         getString(R.string.brave_wallet_network_selection_error,
+                                 networkInfo.chainName),
+                         Toast.LENGTH_SHORT)
+                    .show();
+        }
+        // Add little delay for smooth selection animation
+        new Handler().postDelayed(() -> {
```

Comment:
please use chromium's wrappes for post tasks. Similar to this one https://source.chromium.org/chromium/chromium/src/+/main:android_webview/glue/java/src/com/android/webview/chromium/WebViewContentsClientAdapter.java;l=442?q=WebViewContentsClientAdapter.java&ss=chromium%2Fchromium%2Fsrc